src/cc_catalog_airflow/dags/provider_api_scripts/scan_the_world.py

```diff
@@ -0,0 +1,169 @@
+"""
+Content Provider:       Scan The World
+ETL Process:            Use the API to identify all CC licensed images.
+Output:                 TSV file containing the images and the
+                        respective meta-data.
+Notes:                  https://www.myminifactory.com/api/v2/
+                        Rate limit: 5 requests per second
+"""
+
+import os
+import logging
+from common.requester import DelayedRequester
+from common.storage.image import ImageStore
+from util.loader import provider_details as prov
+
+logging.basicConfig(
+    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s",
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+DELAY = 1
+LIMIT = 100
+RETRIES = 3
+MAX_DESCRIPTION_LENGTH = 2000
+PROVIDER = prov.STW_DEFAULT_PROVIDER
+REQUEST_TYPE = "search"
+ENDPOINT = f"https://www.myminifactory.com/api/v2/{REQUEST_TYPE}"
+API_KEY = os.getenv("SCAN_THE_WORLD_KEY")
+
+delayed_requester = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+HEADERS = {"key": API_KEY}
+
+DEFAULT_QUERY_PARAM = {"cat": 112, "rec_cat": 1, "page": 1, "per_page": LIMIT}
+
+LICENSES = {
+    "Creative Commons - Public Domain": ("cc0", "1.0"),
+    "Creative Commons - Attribution - Noncommercial - ShareAlike":
+    ("by-nc-sa", "2.0"),
+}
+
+
+def main():
+    logger.info("Begin: Stw provider script")
+    condition = True
+    page = 1
+    while condition:
+        query_param = _get_query_param(page=page)
+        object_list = _get_object_list(query_param=query_param)
+        if type(object_list) == list and len(object_list) > 0:
+            _ = _process_object_list(object_list)
+            logger.debug(f"Images till now {image_store.total_images}")
+            page = page + 1
+        else:
+            condition = False
+    image_store.commit()
+    logger.info(f"Total images recieved {image_store.total_images}")
+
+
+def _get_query_param(page, default_query_param=DEFAULT_QUERY_PARAM):
+    query_param = default_query_param.copy()
+    query_param.update(page=page)
+    return query_param
+
+
+def _get_object_list(query_param, headers=HEADERS,
+                     endpoint=ENDPOINT, retries=RETRIES):
+    for r in range(retries):
```

Comment:
Actually I have used get method of DelayedRequester instead of get_response_json so i had to write logic explicitly . I will change the code to use get_response_json method.

Reply:
As I have used get method of DelayedRequester which doesn't take retries as an argument so to change it to get_response_json would require some major changes in both main code and tests .So is it that necessary to take retries as an argument instead of this logic.(i have seen this being used in some existing scripts)